extensions/emmet/src/abbreviationActions.ts

```diff
@@ -161,13 +164,26 @@ function doWrapping(individualLines: boolean, args: any) {
 		let extractedResults = helper.extractAbbreviationFromText(inputAbbreviation);
 		if (!extractedResults) {
 			return Promise.resolve(inPreview);
-		} else if (extractedResults.abbreviation !== inputAbbreviation) {
-			// Not clear what should we do in this case. Warn the user? How?
+		} else if (extractedResults.abbreviation !== inputAbbreviation.replace(/\|[a-zA-Z]+/g, '')) {
+			if (definitive && inPreview) {
+				// If the final text is not a complete abbreviation, we revert any previews made.
+				return revertPreview().then(() => {
+					vscode.window.showErrorMessage('Failed to expand abbreviation ' + inputAbbreviation + '\n' + '  Please check it\'s complete before pressing enter.', 'Ok');
+					return false;
+				});
+			}
+			// If the text is not a complete abbreviation yet, we don't modify the preview.
+			return Promise.resolve(inPreview);
 		}
 
 		let { abbreviation, filter } = extractedResults;
 		if (definitive) {
 			const revertPromise = inPreview ? revertPreview() : Promise.resolve();
+			if (individualLines) {
```

Comment:
You can just assume `individualLines = true` now, so there's no need for this check